File: ai-api-sdk/Java/ktAiApiClient/src/main/java/com/kt/ai/api/client/SttGrpcClient.java
package com.kt.ai.api.client;

import kt.gigagenie.ai.api.STTgRPC;
import kt.gigagenie.ai.api.STTgRPCCallback;
import java.io.FileInputStream;

public class SttGrpcClient {
    public static void main(String[] args) {
        STTgRPC sttGrpcClient = new STTgRPC();
        String clientID = "client-id";
        String clientKey = "client-key";
        String clientSecret = "client-secret";

        STTgRPCCallback onSTTgRPCCallbackListener = new STTgRPCCallback() {
            @Override
            public void onConnectGRPC() {
                System.out.println("onConnectGRPC");

                String sttMode = "long";
                int channel = 1;
                int sampleRate = 16000;
                String sampleFmt = "S16LE";

                sttGrpcClient.startSTT(sttMode, sampleFmt, sampleRate, channel);
            }

            @Override
            public void onSTTResult(String text, String type, float startTime, float endTime) {
                System.out.println("onSTTResult text: " + text);
                System.out.println("onSTTResult type: " + type);
                System.out.println("onSTTResult startTime: " + startTime);
                System.out.println("onSTTResult endTime: " + endTime);
            }

            @Override
            public void onReadySTT(int sampleRate, int channel, String format) {
                System.out.println("onReadySTT >>> sampleRate:" + sampleRate + ", channel:" + channel + ", format:" + format);

                try {
                    boolean isSending = true;
                    FileInputStream inputStream = new FileInputStream("/home/dev/Workspace/GitLab/KtAiApiSDK/Java/ktAiApiClient/src/main/resources/record.pcm");
                    byte[] buffer = new byte[sampleRate];
                    int bytesRead = 0;

                    while (isSending) {
                        bytesRead = inputStream.read(buffer);

                        if (bytesRead == -1) break;
                        Thread.sleep(500);
                        sttGrpcClient.sendAudioData(buffer);
                    }

                    inputStream.close();

                } catch (Exception e) {

                }
            }

            @Override
            public void onStopSTT() {
                System.out.println("onStopSTT");
            }

            @Override
            public void onRelease() {
                System.out.println("onRelease");
            }

            @Override
            public void onError(int errCode, String errMsg) {
                System.out.println("onError >>> errCode:" + errCode + ", errMsg:" + errMsg);
            }
        };
        sttGrpcClient.setSTTgRPCCallback(onSTTgRPCCallbackListener);
        sttGrpcClient.setMetaData(clientKey, clientID, clientSecret);

        sttGrpcClient.connectGRPC();
        try {
            Thread.sleep(15000);
            sttGrpcClient.releaseConnection();
        } catch (Exception e) {

        }
    }
}

File: ai-api-sdk/Node.js/v1.0/ktAiApiClient/sttGrpcClient.js
const STTgRPC = require('./kt-ai-api-sdk/STTgRPC');
const path = require('path');
const fs = require('fs');

var notifier = require('./kt-ai-api-sdk//noti.js');

function sleep(ms) {
    return new Promise((resolve) => {
      setTimeout(resolve, ms);
    });
} 

notifier.on('onError', (message) => {
    console.log('Callback Event onError message ==> ' + JSON.stringify(message));
});

notifier.on('onRelease', (message) => {
    console.log('Callback Event onRelease message ==> ' + message)
});

notifier.on('onReadySTT', async (sampleRate, ch, format) => {
    console.log('Callback Event onReadySTT message ==> ' + sampleRate + ', ' + ch + ', ' + format);
    var sendSize = sampleRate*ch
    for(var i = 0; i <= audioData.length/sendSize; i++) {
        var sendData = audioData.slice(i*sendSize, (i+1)*sendSize);
        gRpcClient.sendAudioData(sendData);
        await sleep(100);
    }
});

notifier.on('onStopSTT', (message) => {
    console.log('Callback Event onStop message ==> ' + message);
});

notifier.on('onSTTResult', (message) => {
    console.log('Callback Event onSTTResult message ==> ' + JSON.stringify(message));
});

const gRpcClient = new STTgRPC();

const clientID = "client-id";
const clientKey = "client-key";
const clientSecret = "client-secret";

gRpcClient.setMetaData(clientKey, clientID, clientSecret);

gRpcClient.connectGRPC();
gRpcClient.startSTT("long", "S16LE", 16000, 1);

pcmPath = path.join(__dirname, "/sample/record.pcm");
var fileData = fs.readFileSync(pcmPath);
var audioData = Buffer.from(fileData, 'binary');

setTimeout(() => {
    gRpcClient.releaseConnection();
    process.exit();
}, 15000);